File: Regression/Inbound/Paper_Form/NRG_Gas.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.select import Select
import time
import xlrd
import os

if os.environ.get('USE_PHANTOM'):
    print("using PhantomJS")

    driver = webdriver.PhantomJS()
    driver.implicitly_wait(5)
    driver.set_window_size(1120, 550)
else:
    print("using Firefox")

workbook = xlrd.open_workbook("./Form_data.xlsx")
worksheet = workbook.sheet_by_name('NRG_regression GAS')

for current_row in range(1,worksheet.nrows):
    brand = worksheet.row(current_row)[0].value
    first_name = worksheet.row(current_row)[1].value
    last_name = worksheet.row(current_row)[2].value
    address = worksheet.row(current_row)[3].value
    city = worksheet.row(current_row)[4].value
    state = worksheet.row(current_row)[5].value
    zipcode = worksheet.row(current_row)[6].value
    area_code = worksheet.row(current_row)[7].value
    prefix = worksheet.row(current_row)[8].value
    last = worksheet.row(current_row)[9].value
    account_type = worksheet.row(current_row)[10].value
    email = worksheet.row(current_row)[11].value
    given_utiity = worksheet.row(current_row)[12].value
    account_number = worksheet.row(current_row)[15].value
    service_ref = worksheet.row(current_row)[16].value
    partner = worksheet.row(current_row)[17].value
    campaign = worksheet.row(current_row)[18].value
    promo = worksheet.row(current_row)[19].value

    driver = webdriver.Firefox()
    driver.get("http://www.pt.energypluscompany.com/myinbound/login.php")

#login
    elem = driver.find_element_by_name("email").send_keys("mpeters@energypluscompany.com")
    elem = driver.find_element_by_name("password").send_keys("energy")
    elem = driver.find_element_by_id("button").click()
    time.sleep(2)

    driver.get("http://www.pt.energypluscompany.com/myinbound/paper.php")
    time.sleep(2)

    elem = driver.find_element_by_id(brand).click()
    time.sleep(3)

#Customer Information
    elem = driver.find_element_by_name("customer_first_name").send_keys(first_name)
    elem = driver.find_element_by_name("customer_last_name").send_keys(last_name)
    elem = driver.find_element_by_name("customer_address1").send_keys(address)
    elem = driver.find_element_by_name("customer_city").send_keys(city)
    elem = driver.find_element_by_name("customer_state")
    for option in elem.find_elements_by_tag_name('option'):
        if option.text == state:
            option.click()
    elem = driver.find_element_by_name("customer_zip5").send_keys(zipcode)
    elem = driver.find_element_by_name("customer_area_code").send_keys(area_code)
    elem = driver.find_element_by_name("customer_prefix").send_keys(prefix)
    elem = driver.find_element_by_name("customer_line_number").clear()
    elem = driver.find_element_by_name("customer_line_number").send_keys(last)
    elem = driver.find_element_by_name("account_type")
    for option in elem.find_elements_by_tag_name('option'):
        if option.text == account_type:
            option.click()
    if 	driver.find_element_by_name("business_name").is_displayed():
        time.sleep(1)
        elem = driver.find_element_by_name("business_name").send_keys(business_name)
        time.sleep(1)
    
#Billing Information
    elem = driver.find_element_by_name("copyinfo").click()
    elem = driver.find_element_by_name("email").send_keys(email)
    if 	driver.find_element_by_id("egr_commodity_gas").is_displayed():
        elem = driver.find_element_by_id("egr_commodity_gas").click()

#Electric
    elem = driver.find_element_by_name("gas_utility")
    for option in elem.find_elements_by_tag_name('option'):
        if option.text == given_utiity:
            option.click()
    elem = driver.find_element_by_name("gas_account_number").send_keys(account_number)

#Order Information
    elem = driver.find_element_by_name("offer_types_gas")
    for option in elem.find_elements_by_tag_name('option'):
    	if option.text == ("Offer Code"):
    		option.click()
    elem = driver.find_element_by_name("partner_code_gas").send_keys(partner)
    elem = driver.find_element_by_name("campaign_code_gas").send_keys(campaign)
    if driver.find_element_by_name("promo_code_gas").is_displayed():
        elem = driver.find_element_by_name("promo_code_gas").send_keys(promo)
    time.sleep(4)

#Vendor ID
    elem = driver.find_element_by_name("vendor_id")
    for option in elem.find_elements_by_tag_name('option'):
        if option.text == ("EPIB"):
            option.click()

# Grab Conformation Code
    elem = driver.find_element_by_name("confCode")
    confCode = elem.text
    print("Passed - NRG_regression Paper - Gas, " + confCode +' for ' + state +' - ' + given_utiity)

    elem = driver.find_element_by_name("submit").click()

    driver.close()

File: Regression/Inbound/Paper_Form/NRG_Gas.rb
require 'rubygems'
require 'watir-webdriver'
require 'win32ole' 
require 'csv'

rows = 2
while  
rows <= 20		#6

excel= WIN32OLE::new("excel.Application")
workbook=excel.Workbooks.Open("C:\\Scripts\\Inbound\\Paper_Form\\Form_Data.xlsx")
worksheet = workbook.worksheets(5)

brand=worksheet.cells(rows,"A").value
first_name=worksheet.cells(rows,"B").value
last_name=worksheet.cells(rows,"C").value
address=worksheet.cells(rows,"D").value
city=worksheet.cells(rows,"E").value
state=worksheet.cells(rows,"F").value
zip=worksheet.cells(rows,"G").value
area_code=worksheet.cells(rows,"H").value
prefix=worksheet.cells(rows,"I").value.to_s
last=worksheet.cells(rows,"J").value.to_s
account_type=worksheet.cells(rows,"K").value
email=worksheet.cells(rows,"L").value
utility=worksheet.cells(rows,"M").value
account_number=worksheet.cells(rows,"P").value.to_s
service_ref=worksheet.cells(rows,"Q").value.to_s
partner=worksheet.cells(rows,"R").value
campaign=worksheet.cells(rows,"S").value
promo=worksheet.cells(rows,"T").value

browser = Watir::Browser.new :firefox
browser.goto 'http://www.pt.energypluscompany.com/myinbound/login.php'

#login
browser.text_field(:name => 'email').set ('mpeters@energypluscompany.com')
browser.text_field(:name => 'password').set ('energy')
browser.button(:value => ' Login').click

browser.goto 'http://www.pt.energypluscompany.com/myinbound/paper.php'
sleep 2

browser.button(:id => brand).click
Watir::Wait.until {browser.text.include? "Confirmation Code"}

wsh = WIN32OLE.new('Wscript.Shell')
wsh.AppActivate('Inbound - Paper Application')

#Customer Information
browser.text_field(:name => 'customer_first_name').set first_name
browser.text_field(:name => 'customer_last_name').set last_name
browser.text_field(:name => 'customer_address1').set address
browser.text_field(:name => 'customer_city').set city
browser.select_list(:name => 'customer_state').select state
browser.text_field(:name => 'customer_zip5').set zip
browser.text_field(:name => 'customer_area_code').set area_code
browser.text_field(:name => 'customer_prefix').set prefix
browser.text_field(:name => 'customer_line_number').set last
browser.select_list(:name => 'account_type').select account_type

if
browser.text_field(:name => 'ssn').visible?
browser.text_field(:name => 'ssn').set ss
end

if
browser.text_field(:name => 'business_name').visible?
browser.text_field(:name => 'business_name').set business_name
end

#Billing Information
browser.checkbox(:name => 'copyinfo').set
browser.checkbox(:name => 'copyinfo').fire_event 'onclick'
browser.text_field(:name => 'email').set email
if
browser.radio(:value => 'gas').visible?
browser.radio(:value => 'gas').set
end

#Electric
browser.select_list(:name => 'gas_utility').select utility
browser.text_field(:name => 'gas_account_number').set account_number

#Order Information
browser.select_list(:name => 'offer_types_gas').select("Offer Code")
browser.text_field(:name => 'partner_code_gas').set partner
browser.text_field(:name => 'campaign_code_gas').set campaign
if 
browser.text_field(:name => 'promo_code_gas').visible?
browser.text_field(:name => 'promo_code_gas').set promo
else
browser.text_field(:name => 'elec_promo_code').set promo
end
sleep 2

#Vendor ID
browser.select_list(:name => 'vendor_id').select ("EPIB")

CSV.open('C:\\Scripts\\Inbound\\Paper_Form\\NRG_Gas_Conf_data.csv', 'ab') do |csv|
csv << [browser.div(:class => 'enrollmentheader').text.slice(-7..-1), partner=worksheet.cells(rows,"V").value, campaign=worksheet.cells(rows,"W").value,
promo=worksheet.cells(rows,"X").value]
end
sleep 2

browser.button(:name => 'submit').click

sleep 1
browser.close 

rows=rows+1

$end
workbook.Save
workbook.Close
end

$end
sleep 2